modules/docs/src/05-design/01-design.md

```diff
@@ -0,0 +1,14 @@
+---
+sidebar_label: General design principles 
+title: General design principles 
+---
+
+Before we dive in the design elements, it is important to state that smithy4s is designed with the following constraints : 
+
+* optimised for number of usecases : smithy4s is **not** an just related to http/json, it can/could be used for a great many things, such as AWS SDKs, CLIs, UIs, etc. Therefore, following the protocol-agnostic nature of smithy, the code generated by smithy4s is not biased towards any protocol or serialisation mechanism, nor any runtime or Scala version. 
```

Comment:
```suggestion
* optimised for a number of usecases : smithy4s is **not** an just related to http/json, it can/could be used for a great many things, such as AWS SDKs, CLIs, UIs, etc. Therefore, following the protocol-agnostic nature of smithy, the code generated by smithy4s is not biased towards any protocol or serialisation mechanism, nor any runtime or Scala version. 
```